packages/calcite-components/src/components/button/button.tsx

```diff
@@ -179,7 +179,7 @@ export class Button
    *
    * @mdn [type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type)
    */
-  @property({ reflect: true }) type = "button";
+  @property({ reflect: true }) type: "submit" | "reset" | "button" = "button";
```

Comment:
(optional) you could also use `LuminaJsx.HTMLElementTags["button"]["type"]` as the type here so that the type updates automatically if more button types are added to DOM in the future (unless you prefer to manually test compatibility with new button types before declaring support for them)